mod-tcp-goodput.md

```diff
@@ -23,25 +28,136 @@ simple data exchange it has some novel features.
 - Selectable congestion control algorithm
 - Parallel Workers (thread support)
 
+## Name
+
+This module is standardized with the name:
+
+```
+tcp-goodput
+```
+
+## Messages
+
+### Info Reply
+
+`tcp-goodput` has no additional information for the client. The `tcp-goodput`
+dictionary MUST be empty.
 
-## Configuration
+E.g.
+
+```
+[]
+  "id" : "hostname=uuid",
+  "seq-rp" : <uint64_t>
+  "modules" : {
+     "tcp-goodput" : { },
+  }
+[]
+```
+
+### Measurement Start Request
 
 ```
 {
-  # per default one TCP transmitter is started, to spawn exactly
+	# per default one TCP transmitter is started, to spawn exactly
 	# to much threads are cores are available use "cores".
 	# Use "threads" if you want to fully utilize all virtual cores,
 	# including hyperthreads.
 	# If the system has several sockets, all sockets are utilized for
 	# "cores" and "threads".
-	"worker" : "1"
+	"streams" : [
+                {
+                    # id, identifying this stream uniquly
+                    "id" = "<uint64_t>",
+                    # port to open and receive data
+                    "port" : "7777",
+                },
+                {
+                    # id, identifying this stream uniquly
+                    "id" = "<uint64_t>",
+                    # port to open and receive data
+                    "port" : "7778",
+                },
+  ]
+}
+```
+
+A minimal example `measurement-start-request` message may look like:
 
-	# payload pattern. Default is zeroized because we want to fullfill
-	# the pipe and offload as much as possible. 
-	"payload-pattern" : "zeroized"
+```
+{
+  "id" : "hostname=uuid",
+  "measurement-id" : "<uint64_t>"
+  "seq" : "<uint64_t>"
+  "secret" : "<string>"
+  
+  "measurement" : {
+        "name" : "tcp-goodput"
+        "type" : "module"
+        "configuration" : {
+            "streams" : [
+                {
+                    # id, identifying this stream uniquly
+                    "id" = "<uint64_t>",
+                    # port to open and receive data
+                    "port" : "7777",
+                },
+                {
+                    # id, identifying this stream uniquly
+                    "id" = "<uint64_t>",
+                    # port to open and receive data
+                    "port" : "7778",
+                },
```

Comment:
snapshot length is missing here...used in maparo pulser...

Reply:
Right, size should be more appropriate for this. IMHO